sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/ProxyOptions.java

```diff
@@ -67,6 +72,43 @@ public ProxyOptions(ProxyAuthenticationType authentication, Proxy proxyAddress,
         }
     }
 
+    /**
+     * Attempts to load a proxy from the configuration or {@code null} if no proxy was found in the configuration.
+     *
+     * @param configuration The {@link Configuration} that is used to load proxy configurations from the environment. If
+     * {@code null} is passed then {@link Configuration#getGlobalConfiguration()} will be used. If {@link
+     * Configuration#NONE} is passed {@link IllegalArgumentException} will be thrown.
+     * @return A {@link ProxyOptions} reflecting a proxy loaded from the environment, if no proxy is found {@code null}
+     * will be returned.
+     * @throws IllegalArgumentException If {@code configuration} is {@link Configuration#NONE} or configuration is invalid.
```

Comment:
When is a configuration invalid other than being `Configuration.NONE`? Or is this meant to be the proxy contained in the configuration is invalid?